app/lib/applets/timetable/student/student_timetable_view.dart

```diff
@@ -134,12 +139,14 @@ class _StudentTimetableViewState extends State<StudentTimetableView> {
                   minDate: DateTime.now(),
                   maxDate: DateTime.now().add(const Duration(days: 7)),
                   controller: controller,
-                  onViewChanged: (_) {
-                    logger.i(
-                        "Setting \"currentTimeTableView\" to \"${controller.view}\"");
-                    accountDatabase.kv
-                        .set("currentTimeTableView", "${controller.view}");
-                    _loadCalendarView();
+                  onViewChanged: (_) async {
+                    if (await _getCalendarView() != "${controller.view}") {
+                      logger.i(
+                          "Setting \"currentTimeTableView\" to \"${controller.view}\"");
+                      accountDatabase.kv
```

Comment:
Shouldn't the settings from the CombinedAppletBuilder be used, as it is specifically for this applet? (Like I did wrong in the prior PR)?

Reply:
Yes. This also looks like a setting that should be applet+user specific. The "accountDatabase" is shared between different user accounts